src/wp-includes/block-editor.php

```diff
@@ -0,0 +1,335 @@
+	}
+
+	return $block_categories;
+}
+
+/**
+ * Gets the list of allowed block types to use in the block editor.
+ *
+ * @since 5.8.0
+ *
+ * @param string $editor_name The name of the editor (e.g. 'post-editor').
+ *
+ * @return bool|array Array of block type slugs, or boolean to enable/disable all.
+ */
+function get_allowed_block_types( $editor_name ) {
+	$allowed_block_types = true;
+
+	/**
+	 * Filters the allowed block types for the given editor, defaulting to true (all
+	 * registered block types supported).
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool|array $allowed_block_types Array of block type slugs, or
+	 *                                        boolean to enable/disable all.
+	 */
+	$allowed_block_types = apply_filters( "allowed_block_types_{$editor_name}", $allowed_block_types );
+	if ( 'post-editor' === $editor_name ) {
+		$post = get_post();
+
+		/**
+		 * Filters the allowed block types for the editor, defaulting to true (all
+		 * block types supported).
+		 *
+		 * @since 5.0.0
+		 * @deprecated 5.8.0 The hook transitioned to support also screens that don't contain $post instance.
+		 *
+		 * @param bool|array $allowed_block_types Array of block type slugs, or
+		 *                                        boolean to enable/disable all.
+		 * @param WP_Post    $post                The post resource data.
+		 */
+		$allowed_block_types = apply_filters_deprecated( 'allowed_block_types', array( $allowed_block_types, $post ), '5.8.0', "allowed_block_types_{$editor_name}" );
+	}
+
+	return $allowed_block_types;
+}
+
+/**
+ * Returns the default block editor settings.
+ *
+ * @since 5.8.0
+ *
+ * @return array The default block editor settings.
+ */
+function get_default_block_editor_settings() {
+	// Media settings.
+	$max_upload_size = wp_max_upload_size();
+	if ( ! $max_upload_size ) {
+		$max_upload_size = 0;
+	}
+
+	/** This filter is documented in wp-admin/includes/media.php */
+	$image_size_names = apply_filters(
+		'image_size_names_choose',
+		array(
+			'thumbnail' => __( 'Thumbnail' ),
+			'medium'    => __( 'Medium' ),
+			'large'     => __( 'Large' ),
+			'full'      => __( 'Full Size' ),
+		)
+	);
+
+	$available_image_sizes = array();
+	foreach ( $image_size_names as $image_size_slug => $image_size_name ) {
+		$available_image_sizes[] = array(
+			'slug' => $image_size_slug,
+			'name' => $image_size_name,
+		);
+	}
+
+	$default_size       = get_option( 'image_default_size', 'large' );
+	$image_default_size = in_array( $default_size, array_keys( $image_size_names ), true ) ? $image_default_size : 'large';
+
+	$image_dimensions = array();
+	$all_sizes        = wp_get_registered_image_subsizes();
+	foreach ( $available_image_sizes as $size ) {
+		$key = $size['slug'];
+		if ( isset( $all_sizes[ $key ] ) ) {
+			$image_dimensions[ $key ] = $all_sizes[ $key ];
+		}
+	}
+
+	$editor_settings = array(
+		'alignWide'              => get_theme_support( 'align-wide' ),
+		'allowedBlockTypes'      => true,
```

Comment:
:wave: is this meant to be here given that we also set it [here](https://github.com/WordPress/wordpress-develop/pull/1118/files#diff-edfc853e670c7bfc5a838dc74f75ba5d8b95cb4adbddf0644d9227bcb692e344R237)?

Reply:
Yes, the related discussion is here:
https://github.com/WordPress/wordpress-develop/pull/1118#discussion_r605526076

It's something that I'm not 100% sure about so all feedback is valuable.